test/Test/Database/LSMTree/Internal/BloomFilter.hs

```diff
@@ -31,3 +40,28 @@ roundtrip_prop (Positive (Small hfN)) (limitBits -> bits) ws =
 
 limitBits :: Word64 -> Word64
 limitBits b = b .&. 0xffffff
+
+prop_total_deserialisation :: [Word32] -> Property
+prop_total_deserialisation word32s =
+    case bloomFilterFromSBS (SBS.SBS ba) of
+      Left err -> label err $ property True
+      Right bf -> label "parsed successfully" $ property $
+        -- Just forcing the filter is not enough (e.g. the bit vector might
+        -- point outside of the byte array).
+        bf `deepseq` BF.bloomInvariant bf
+  where
+    !(ByteArray ba) = byteArrayFromList word32s
+
+-- Length is in Word64s. A large length would require significant amount of
+-- memory, so we make it 'Small'.
+prop_total_deserialisation_whitebox :: Word32 -> Small Word32 -> Property
+prop_total_deserialisation_whitebox hsn (Small len64) =
+      forAll (vector (fromIntegral len64 * 2)) $ \word32s ->
+        prop_total_deserialisation (prefix <> word32s)
+  where
+    prefix =
+      [ 1 {- version -}
+      , hsn
+      , unsafeShiftL len64 6         -- len64 * 64 (lower 32 bits)
+      , unsafeShiftR len64 (32 - 6)  -- len64 * 64 (upper 32 bits)
```

Comment:
```suggestion
      , unsafeShiftR len64 (32 - 6)  -- len64 / 64 (upper 32 bits)
```

Reply:
No, it's multiplication (left shift), but since we only take the upper 32 bits, it turns into an overall right shift.

Reply:
Hmm okay I think I misunderstood the calculation, but it's clear now